### ISSUE TITLE
Mod makes it impossible to noclip in spectator mode

### ISSUE THREAD

**Issue by axd1x8a** [Phelms215/JoinLeaveMessages#2]
As the title says, in [this](https://modrinth.com/mod/joinleavemessages/version/GcnqzymS) version you won't be able to go through blocks in spectator mode

**Comment by Wenz-jam** [comment]
This might be related to the issue — just pointing it out in case it helps.
https://github.com/Phelms215/JoinLeaveMessages/blob/3b4b69a659b8d4b971e81e18abe7e0157427b9e9/forge/1.20.2/src/main/java/com/networkglitch/joinleavemessages/mixin/PlayerLeaveMixin.java#L43-L51
https://github.com/Phelms215/JoinLeaveMessages/blob/3b4b69a659b8d4b971e81e18abe7e0157427b9e9/forge/1.19.3/src/main/java/com/networkglitch/joinleavemessages/mixin/PlayerLeaveMixin.java#L43-L51
https://github.com/Phelms215/JoinLeaveMessages/blob/3b4b69a659b8d4b971e81e18abe7e0157427b9e9/forge/1.18.2/src/main/java/com/networkglitch/joinleavemessages/mixin/PlayerLeaveMixin.java#L45-L53

**Comment by Camawama** [comment]
Yeah I also noticed this issue with spectator mode. It also allows spectators to block non-spectators from placing blocks. Very broken! It also makes spectators no longer invisible to normal players.
